app/controllers/chat_channels_controller.rb

```diff
@@ -262,6 +262,38 @@ def change_status
     render json: success_json
   end
 
+  def destroy
+    params.require(:chat_channel_id)
+    params.require(:channel_name_confirmation)
+
+    chat_channel = ChatChannel.find_by(id: params[:chat_channel_id])
+    raise Discourse::NotFound if chat_channel.blank?
+
+    guardian.ensure_can_delete_chat_channel!
+
+    if chat_channel.name.downcase != params[:channel_name_confirmation].downcase
```

Comment:
```suggestion
    if chat_channel.name.downcase != params[:channel_name_confirmation]
```

I don't think we need to downcase the param on the server side since we're expecting a name that is already lowercased?

Reply:
IMO the server should do the same thing as the frontend, you could  send a DELETE request independently from the frontend

Reply:
But we're not exposing this as an API outside of use for the client so I don't think we need to handle cases where people send a DELETE request independently from the frontend. I do wonder if it is even necessary for us to be comparing the channel's name on the server side. If it is, we can just drop the `chat_channel_id` params because both are unique columns that can be used to identify the same resource.

Reply:
I mean...by default we are exposing everything. IMO we should just leave this as is, it adds an extra layer of confirmation and I would rather look up by ID than name.

Reply:
And by default exposing everything I mean there is no way to stop anyone just using this as is. I will check what github POSTS when you delete a repo, and see if they send through the confirmation, and if not then I will remove it here

Reply:
Yea I do wonder what Github does in this case too. I've always thought having to type the name of the repo is just to make sure that the trigger to delete the repository is an intentional one.

Reply:
```
_method: delete
authenticity_token: w8gErekdNAu5gEeIaYMyVmkyo73PLra5dhpD9Ex6LEIT4V9yevmdjHVnSPkMWKCtUS5-eRqYKGhHo1hNnIp29Q
repository: blahblahdeletetest
sudo_referrer: https://github.com/martin-brennan/blahblahdeletetest/settings
user_id: martin-brennan
verify: martin-brennan/blahblahdeletetest
```

GitHub does the same thing with the verify param :)

Reply:
OK :ok_hand: Let's keep it as it is then :)